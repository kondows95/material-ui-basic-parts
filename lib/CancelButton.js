var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { Box } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useIntl } from 'react-intl';
import { getLeftButtonStyle } from './styles';
import MyButton from './MyButton';
var useStyles = makeStyles(function (theme) { return ({
    item: getLeftButtonStyle(theme),
}); });
var CancelButton = function (props) {
    var classes = useStyles();
    var formatMessage = useIntl().formatMessage;
    return (React.createElement(Box, { className: classes.item },
        React.createElement(MyButton, __assign({}, props, { label: formatMessage({ id: 'MUIBP.cancel', defaultMessage: 'Cancel' }), variant: "contained", fullWidth: true, "data-testid": "cancel-button" }))));
};
export default CancelButton;
