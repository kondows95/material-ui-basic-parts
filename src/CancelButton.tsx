import React from 'react';
import { Button, Box, Theme } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useIntl } from 'react-intl';

const useStyles = makeStyles((theme: Theme) => ({
    item: {
        [theme.breakpoints.down('xs')]: {
            padding: theme.spacing(2, 0, 1), //TXB
        },
        [theme.breakpoints.up('sm')]: {
            padding: theme.spacing(2, 1, 1, 0), //TRBL
        },
    },
}));

interface Props {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: string]: any;
}

const CancelButton: React.FC<Props> = (props) => {
    const classes = useStyles();
    const { formatMessage } = useIntl();

    return (
        <Box className={classes.item}>
            <Button {...props} variant="contained" fullWidth data-testid="cancel-button">
                {formatMessage({ id: 'MUIBP.cancel', defaultMessage: 'Cancel' })}
            </Button>
        </Box>
    );
};
export default CancelButton;
