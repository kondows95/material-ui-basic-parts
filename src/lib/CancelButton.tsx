import React from 'react';
import { Box, Theme } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useIntl } from 'react-intl';
import { getLeftButtonStyle } from './styles';
import MyButton from './MyButton';

const useStyles = makeStyles((theme: Theme) => ({
    item: getLeftButtonStyle(theme),
}));

interface Props {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: string]: any;
    label?: string;
}

interface CopyProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    [key: string]: any;
}

const CancelButton: React.FC<Props> = (props) => {
    const classes = useStyles();
    const { formatMessage } = useIntl();

    const copyProps: CopyProps = {};
    for (const key of Object.keys(props)) {
        if (key !== 'label') {
            copyProps[key] = props[key];
        }
    }

    const msgId = 'MUBP.CancelButton';
    return (
        <Box className={classes.item}>
            <MyButton
                {...copyProps}
                label={props.label || formatMessage({ id: msgId, defaultMessage: 'Cancel' })}
                variant="contained"
                fullWidth
                data-testid={msgId}
            />
        </Box>
    );
};
export default CancelButton;
