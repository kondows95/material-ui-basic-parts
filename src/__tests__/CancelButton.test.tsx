import React from 'react';
import TestLib from 'react-testing-library-wrapper';
import { IntlProvider } from 'react-intl';
import CancelButton from '../lib/CancelButton';

let tLib: TestLib;
let onClick: jest.Mock;

beforeEach(() => {
    onClick = jest.fn();
});

describe('Basic Usage', () => {
    beforeEach(() => {
        tLib = new TestLib(
            (
                <IntlProvider locale="en" messages={{ 'MUIBP.cancel': 'MyLabel' }}>
                    <CancelButton onClick={onClick} className="MyClass" />
                </IntlProvider>
            )
        );
    });

    it('You can set button label by Intl', () => {
        expect(tLib.render.container.textContent).toBe('MyLabel');
    });

    it('You can add onClick event', () => {
        tLib.click('cancel-button');
        expect(onClick).toHaveBeenCalled();
    });

    it('You can add className', () => {
        expect(tLib.get('cancel-button').getAttribute('class')).toContain('MyClass');
    });

    it('Snapshot', () => {
        expect(tLib.render.asFragment()).toMatchSnapshot();
    });
});

it('Default label is Cancel', () => {
    tLib = new TestLib(
        (
            <IntlProvider locale="en">
                <CancelButton onClick={onClick} />
            </IntlProvider>
        )
    );
    expect(tLib.render.container.textContent).toContain('Cancel');
});

it('You can set label', () => {
    tLib = new TestLib(
        (
            <IntlProvider locale="en">
                <CancelButton label="MyLabel" onClick={onClick} />
            </IntlProvider>
        )
    );
    expect(tLib.render.container.textContent).toContain('MyLabel');
});
