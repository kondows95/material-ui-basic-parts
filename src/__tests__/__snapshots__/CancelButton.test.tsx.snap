// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Usage Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-141 makeStyles-item-139"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MyClass MuiButton-fullWidth"
      data-testid="cancel-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        MyLabel
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Check Default Values Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-233 makeStyles-item-231"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-fullWidth"
      data-testid="cancel-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Cancel
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;

exports[`Expansion Usage Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-233 makeStyles-item-231"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-fullWidth"
      data-testid="cancel-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Cancel
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
